frontend/src/scenes/plugins/pluginsLogic.ts

```diff
@@ -528,6 +573,57 @@ export const pluginsLogic = kea<pluginsLogicType<PluginForm>>({
             (showingMetricsPluginId, installedPlugins) =>
                 showingMetricsPluginId ? installedPlugins.find((plugin) => plugin.id === showingMetricsPluginId) : null,
         ],
+        filteredUninstalledPlugins: [
+            (s) => [s.searchTerm, s.uninstalledPlugins],
+            (searchTerm, uninstalledPlugins) =>
+                searchTerm
+                    ? uninstalledPlugins.filter((plugin) =>
+                          plugin.name.toLowerCase().includes(searchTerm.toLowerCase())
+                      )
+                    : uninstalledPlugins,
+        ],
+        filteredDisabledPlugins: [
+            (s) => [s.searchTerm, s.disabledPlugins],
+            (searchTerm, disabledPlugins) =>
+                searchTerm
+                    ? disabledPlugins.filter((plugin) => plugin.name.toLowerCase().includes(searchTerm.toLowerCase()))
+                    : disabledPlugins,
+        ],
+        filteredEnabledPlugins: [
+            (s) => [s.searchTerm, s.enabledPlugins],
+            (searchTerm, enabledPlugins) =>
+                searchTerm
+                    ? enabledPlugins.filter((plugin) => plugin.name.toLowerCase().includes(searchTerm.toLowerCase()))
+                    : enabledPlugins,
+        ],
+        filteredPluginsNeedingUpdates: [
+            (s) => [s.searchTerm, s.pluginsNeedingUpdates],
+            (searchTerm, pluginsNeedingUpdates) =>
+                searchTerm
+                    ? pluginsNeedingUpdates.filter((plugin) =>
+                          plugin.name.toLowerCase().includes(searchTerm.toLowerCase())
+                      )
+                    : pluginsNeedingUpdates,
+        ],
+        sortableEnabledPlugins: [
+            (s) => [s.filteredEnabledPlugins],
+            (filteredEnabledPlugins) => {
+                return filteredEnabledPlugins.filter(
+                    (plugin) =>
+                        !plugin.capabilities ||
+                        plugin.capabilities.methods.includes('processEvent') ||
+                        plugin.capabilities.methods.includes('processEventBatch')
```

Comment:
This crashed the Plugins page for me when I tried to enable a code editor plugin, because its `capabilities` object didn't have `methods`.

Reply:
Good catch!